src/directives/NgGrid.ts

```diff
@@ -1016,28 +1050,35 @@ export class NgGrid implements OnInit, DoCheck, OnDestroy {
 		while (this._hasGridCollision(pos, dims) || !this._isWithinBounds(pos, dims, true)) {
 			if (this._hasGridCollision(pos, dims)) {
 				const collisions: NgGridItem[] = this._getCollisions(pos, dims);
-
-				switch (this.cascade) {
-					case 'up':
-					case 'down':
-					default:
-						pos.row = Math.max.apply(null, collisions.map((item: NgGridItem) => item.row + item.sizey));
-						break;
-					case 'left':
-					case 'right':
-						pos.col = Math.max.apply(null, collisions.map((item: NgGridItem) => item.col + item.sizex));
-						break;
+				let fixDirection: NgConfigFixDirection = this._itemFixDirection;

```

Comment:
The method mentioned earlier could be reused here to calculate the computed fix direction.